plugins/discovery-gce/src/main/java/org/elasticsearch/plugin/discovery/gce/GceDiscoveryPlugin.java

```diff
@@ -118,9 +117,9 @@ public static boolean isDiscoveryAlive(Settings settings, ESLogger logger) {
             return false;
         }
 
-        if (checkProperty(GceComputeService.Fields.PROJECT, settings.get(GceComputeService.Fields.PROJECT), logger) == false ||
-                checkProperty(GceComputeService.Fields.ZONE, settings.getAsArray(GceComputeService.Fields.ZONE), logger) == false) {
-            logger.debug("one or more gce discovery settings are missing. " +
+        if (checkProperty(settings.get(GceComputeService.Fields.PROJECT)) == false ||
```

Comment:
if someone configures discovery.type to GCE and these properties are missing, we should throw a hard exception no? why do we only warn?